packages/react-app/src/lib/RainbowWeb3authConnector.ts

```diff
@@ -0,0 +1,65 @@
+import { CHAIN_NAMESPACES } from '@web3auth/base'
+import { Web3AuthNoModal } from '@web3auth/no-modal'
+import { OpenloginAdapter } from '@web3auth/openlogin-adapter'
+import { Web3AuthConnector } from '@web3auth/web3auth-wagmi-connector'
+import { Chain } from 'wagmi'
+
+function createWeb3AuthInstance(chains: Chain[], clientId: string) {
+	// Create Web3Auth Instance
+	const web3AuthInstance = new Web3AuthNoModal({
+		clientId,
+		chainConfig: {
+			chainNamespace: CHAIN_NAMESPACES.EIP155,
+			chainId: '0x' + chains[0].id.toString(16),
```

Comment:
`chain[0]` is expected to be Gnosis Chain? in that case we should ask for Gnosis Chain explicitly